manager/dispatcher/dispatcher.go

```diff
@@ -387,6 +394,20 @@ func (d *Dispatcher) updateNode(nodeID string, description *api.NodeDescription)
 	return nil
 }
 
+// gets the node IP from the context of a grpc call
+func nodeIpFromContext(ctx context.Context) (string, error) {
+	if p, ok := peer.FromContext(ctx); !ok {
```

Comment:
This won't work correctly when the node is connected via a follower manager's proxy.

Instead, use:

```go
nodeInfo, err := ca.RemoteNode(ctx)
if err != nil {
        return err
}
addr, _, err := net.SplitHostPort(nodeInfo.RemoteAddr)
```